src/erasure-code/ErasureCode.cc

```diff
@@ -60,6 +60,28 @@ int ErasureCode::minimum_to_decode(const set<int> &want_to_read,
   return 0;
 }
 
+int ErasureCode::minimum_to_decode2(const set<int> &want_to_read,
+                                   const set<int> &available_chunks,
+                                   map<int, list<pair<int, int>>> *minimum)
```

Comment:
can we replace the list<pair<int,int>> with vector<pair<int,int>>?  it'll be a zillion times faster to avoid the memory allocations.